lib/src/package_analysis/common.dart

```diff
@@ -1,12 +1,204 @@
-import 'package:analyzer/dart/analysis/analysis_context.dart';
-import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'dart:convert';
+import 'dart:io';
 
-abstract class PackageAnalysisContext {
-  AnalysisContextCollection get analysisContextCollection;
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/file_system/file_system.dart' as resource;
+import 'package:pana/pana.dart';
+import 'package:path/path.dart' as path;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart' hide Pubspec;
+
+const indentedEncoder = JsonEncoder.withIndent('  ');
 
-  AnalysisContext contextFor(String path) =>
-      analysisContextCollection.contextFor(path);
+abstract class PackageAnalysisContext {
```

Comment:
add field `packageName` - root package name, for error messages
add field `dependencies` with type Map<String,Version> that maps from (hosted) dependency name to installed version
also add field `pubspec` and field `packagePath`